### PR TITLE
Use pathlib in testing module's iptestcontroller.py

### PR DESCRIPTION
**Pull request by nicksspirit**
Made use of pathlib's Path object to improve iptestcontroller.py

### CODE REVIEW

**Comment by OdinTech3** [comment]
Fixes https://github.com/ipython/ipython/issues/12515

**Comment by OdinTech3** [comment]
Is there more I need to do for this PR?

**Comment by Carreau** [comment]
> Is there more I need to do for this PR?

Sorry there have been many PRs during hacktoberfest and I'm struggling to review them all. 
The test are failing on windows, it seem like some subprocess API is expecting only str and get Path.

File: IPython/testing/iptestcontroller.py [hunk 1]
```diff
@@ -173,26 +175,26 @@ def will_run(self):
...
+            "[run]\n" "data_file = {data_file}\n" "source =\n" "  {source}\n"
+        ).format(
+            data_file=Path(f".coverage.{self.section}").absolute(),
+            source="\n  ".join(sources),
+        )
+        config_file = Path(self.workingdir.name) / ".coveragerc"
+        config_file.touch(exist_ok=True)
+        config_file.write_text(coverage_rc)
+
+        self.env["COVERAGE_PROCESS_START"] = config_file.resolve()
```
Comment (Carreau) [inline IPython/testing/iptestcontroller.py:197]: I think
```suggestion
        self.env["COVERAGE_PROCESS_START"] = str(config_file.resolve())
```

Comment (OdinTech3) [reply IPython/testing/iptestcontroller.py]: Oh okay makes sense.

**Comment by Carreau** [comment]
I've pushed a fix, it seem to be working on windows, waiting for linux build to pass.

**Comment by OdinTech3** [comment]
Sweet!! Looks like your fix lead to CI passing.

**Pull request merged**